fl4health/clients/basic_client.py

```diff
@@ -88,7 +84,7 @@ def fit(self, parameters: NDArrays, config: Config) -> Tuple[NDArrays, int, Dict
 
         if local_epochs is not None:
             _, metrics = self.train_by_epochs(local_epochs, current_server_round)
-            local_steps = self.num_train_samples * local_epochs  # total steps over training round
+            local_steps = len(self.train_loader) * local_epochs  # total steps over training round
```

Comment:
@jewelltaylor: Correct me if I'm wrong, but I believe the LHS was a bug that popped up somewhere? The number of steps is the number of batches in the loader times the number of epochs rather than the number of sample times the number of epochs I think? So I fixed it here.

Reply:
Yeah, I must have accidentally computed the total samples saw through all the epochs in a given training round instead of the number of steps. Good catch!